hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -264,61 +277,73 @@ async def _place_order(
         price: Decimal,
         **kwargs,
     ) -> Tuple[str, float]:
-        side = api.Side.S_BID if trade_type == TradeType.BUY else api.Side.S_ASK
-        type = api.OrderType.OT_LIMIT if order_type == OrderType.LIMIT else api.OrderType.OT_MARKET
+        blxr_order_type = order_type_to_blxr_order_type(order_type)
+        blxr_side = trade_type_to_side(trade_type)
 
         tokens = trading_pair.split("-")
         base = tokens[0]
         quote = tokens[1]
 
-        # this is temporarily hard coded to a single solana wallet
-        base_addr = CONSTANTS.TOKEN_PAIR_TO_WALLET_ADDR[base]
-        quote_addr = CONSTANTS.TOKEN_PAIR_TO_WALLET_ADDR[quote]
-        payer_address = base_addr if side == api.Side.S_ASK else quote_addr
+        payer_address = self._payer_address(base, quote, blxr_side)
+        open_orders_address = ""
+        if trading_pair in self._open_orders_address_mapper:
+            open_orders_address = self._open_orders_address_mapper[trading_pair]
 
         blxr_client_order_id = convert_hummingbot_to_blxr_client_order_id(order_id)
-        self._hummingbot_to_solana_id[order_id] = blxr_client_order_id
+        self._order_id_mapper[order_id] = blxr_client_order_id
 
-        submit_order_response = await self._provider_1.submit_order(
+        await self._testnet_provider.wait_connect()
```

Comment:
Since we're indicating this status everywhere else are we sure this wait call is needed? Fine if it; just thought that since we have stuff like `status_dict` hummingbot would handle that intelligently. Something to try to remove if at all possible.

Reply:
From what I see, it is not exactly clear which functions are called before and after the status_dict is verified. For example, it seems `_initialize_token_accounts` is called before.

I could try to find that out, but it seems easier to just `wait_connect` for all of them.